src/main/java/com/firebolt/jdbc/statement/FireboltStatement.java

```diff
@@ -203,7 +206,45 @@ protected int executeUpdate(List<StatementInfoWrapper> sql) throws SQLException
 		return 0;
 	}
 
-	@Override
+	public void executeAsync(String sql) throws SQLException {
+		StatementInfoWrapper query = StatementUtil.parseToStatementInfoWrappers(sql).get(0);
+        createValidator(query.getInitialStatement(), connection).validate(query.getInitialStatement());
+        synchronized (statementsToExecuteLabels) {
+            statementsToExecuteLabels.add(query.getLabel());
+        }
+        if (isStatementNotCancelled(query)) {
+            runningStatementLabel = determineQueryLabel(query);
+            synchronized (this) {
+                validateStatementIsNotClosed();
+            }
+            try {
+                log.debug("Executing the statement with label {} : {}", query.getLabel(),
+                        sanitizeSql(query.getSql()));
+                if (query.getType() != StatementType.NON_QUERY) {
+                    throw new FireboltException("Only non-queries are supported for async statements");
```

Comment:
Technically "SET x=y" is also a non-query. Maybe let's rephrase this to make sure it's clear what types of queries are supported.